featuretools/utils/recommend_primitives.py

```diff
@@ -0,0 +1,242 @@
+        Then, numeric fields are checked for skewness. Depending on how skew a column is `square_root` or `natural_logarithm` will be recommended.
+        Lastly if `include_time_series_primitives` is specified as `True`, `Lag` will always be recommended,
+        as well as all Rolling and Expanding primitives if numeric columns are present.
+
+    Args:
+        entityset (EntitySet): EntitySet that only contains one dataframe.
+        include_time_series_primitives (bool): Whether or not time-series primitives should be considered.
+        excluded_primitives (List[str]): List of transform primitives to exclude from recommendations.
+
+    Note:
+        The main objective of this function is to recommend primitives that could potentially provide important features to the modeling process.
+        Non-numeric primitives do a great job in mainly serving as a way to extract information from origin features that may essentially be meaningless by themselves (e.g., NaturalLaguange, Datetime, LatLong).
+        That is why they are the main focus of this function. Numeric transform primitives are very case-by-case dependent and therefore it is hard to mathematically quantify which should be recommended.
+        Therefore, we decided to only look at transform primitives that address skewed numeric columns as this is a standard and quantifiable transformation step. The only exception to this rule being
+        for time series problems. Because there are so few primitives that are only applicable for time series, we recommend users try all of them.
+
+    Note:
+        This function currently only works for single table and will only recommend transform primitives.
+    """
+    es_dataframe_list = list(entityset.dataframe_dict)
+    if len(es_dataframe_list) == 0:
+        raise IndexError("No DataFrame in EntitySet found. Please add a DataFrame.")
+    if len(es_dataframe_list) > 1:
+        raise IndexError(
+            "Multi-table EntitySets are currently not supported. Please only use a single table EntitySet.",
+        )
+
+    target_dataframe_name = es_dataframe_list[0]
+
+    recommended_primitives = set()
+
+    if not include_time_series_primitives:
+        excluded_primitives += TIME_SERIES_PRIMITIVES
+
+    all_trans_primitives = get_transform_primitives()
+    selected_trans_primitives = [
+        p for name, p in all_trans_primitives.items() if name not in excluded_primitives
+    ]
+
+    valid_primitive_names = [
+        prim.name
+        for prim in get_valid_primitives(
+            entityset,
+            target_dataframe_name,
+            1,
+            selected_trans_primitives,
+        )[1]
+    ]
+
+    recommended_primitives.update(
+        _recommend_non_numeric_primitives(
+            entityset,
+            target_dataframe_name,
+            valid_primitive_names,
+        ),
+    )
+    skew_numeric_primitives = set(["square_root", "natural_logarithm"])
```

Comment:
In terms of separation of concerns, would it make more sense for the `_recommend_skew_numeric_primitives` function to contain this list? We have this list here and the rest of the logic in the other function, which means you need to look at two functions to understand what gets recommended. Same thing for the time series logic.

Just thinking maybe it would make more sense for us to always call those functions and have the logic for what to do based on the inputs all contained within those functions.

Reply:
In terms of the first comment, yeah this happened in `latlong_to_city` because of an issue with its reverse geocoder. I put this in place just in case there are primitives with issues now or in the future.

Reply:
As for the second comment, done. It doe make it look cleaner. The time series stuff is a one liner so I didn't put it in it's own function. lmk what you think

Reply:
I like it!